.travis.yml

```diff
@@ -16,10 +16,14 @@ cache:
     - node_modules
     - .eslintcache
 
+before_install:
+  - dpkg --compare-versions `npm -v` ge 6.0 || npm i -g npm@latest
+
 install:
-  # --save flag needed b/c of https://github.com/npm/npm/issues/16901
-  - npm update --save false
-  - npm prune
+  - if [[ "$TRAVIS_BRANCH" = "greenkeeper/"* ]]; then npm install; else npm ci; fi
+
+before_script:
+  - greenkeeper-lockfile-update
```

Comment:
What does this do?

Reply:
Helps deal with `package-lock.json` files now that the file is committed to the repo.

This what happens:
1. Greenkeeper makes the normal PR updating `package.json`
2. Travis push build passes because it runs `npm install`, Travis PR build fails because it runs `npm ci` and `package.json` and `package-lock.json` are inconsistent
3. `greenkeeper-lockfile-update` runs on the push build and updates the `package-lock.json`, `greenkeeper-lockfile-upload` commits the `package-lock.json` and pushes it 
4. The commit pushed by `greenkeeper-lockfile-upload` passes both Travis builds as `package-lock.json` and `package.json` are now consistent

Reply:
do the commits created by `greenkeeper-lockfile-upload` cause a new travis build to kick off?

Reply:
yes they do and no they get pushed to this branch